### PR TITLE
[SPARK-36046][SQL] Support new functions make_timestamp_ntz and make_timestamp_ltz

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Support new functions make_timestamp_ntz and make_timestamp_ltz
Syntax:
* `make_timestamp_ntz(year, month, day, hour, min, sec)`: Create local date-time from year, month, day, hour, min, sec fields
* `make_timestamp_ltz(year, month, day, hour, min, sec[, timezone])`: Create current timestamp with local time zone from year, month, day, hour, min, sec and timezone fields

### Why are the changes needed?

As the result of `make_timestamp` become consistent with the SQL configuration `spark.sql.timestmapType`, we need these two new functions to construct timestamp literals. They align to the functions [`make_timestamp` and `make_timestamptz`](https://www.postgresql.org/docs/9.4/functions-datetime.html) in PostgreSQL

### Does this PR introduce _any_ user-facing change?

Yes, two new datetime functions: make_timestamp_ntz and make_timestamp_ltz.

### How was this patch tested?

End-to-end tests.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@gengliangwang Could you look at:
```
[info] - Check schemas for expression examples *** FAILED *** (609 milliseconds)
[info]   364 did not equal 366 Expected 364 blocks in result file but got 366. Try regenerating the result files. (ExpressionsSchemaSuite.scala:161)
[info]   org.scalatest.exceptions.TestFailedException:
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala [hunk 1]
```diff
@@ -552,6 +552,8 @@ object FunctionRegistry {
     expression[TimeWindow]("window"),
     expression[MakeDate]("make_date"),
     expression[MakeTimestamp]("make_timestamp"),
+    expression[MakeTimestampNTZ]("make_timestamp_ntz", true),
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:555]: If `setAlias` is set `true`, it means that an expression has multiple names but in your case `MakeTimestampNTZ` has only `make_timestamp_ntz`. Or `setAlias` has another meaning?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala [hunk 2]
```diff
@@ -2284,6 +2284,128 @@ case class MakeDate(
     copy(year = newFirst, month = newSecond, day = newThird)
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(year, month, day, hour, min, sec[, timezone]) - Create local date-time from year, month, day, hour, min, sec and timezone fields. ",
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala:2289]: Does the expression accept a time zone? I don't see that ...

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala (hunk 1 above)
Comment (gengliangwang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:555]: It means we don't need to override the `prettyName` method.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala (hunk 2 above)
Comment (gengliangwang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala]: Updated. Sorry about that.

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master/3.2.
Thank you, @gengliangwang .

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:555]: It's better to implement `prettyName`. The alias here is implemented via `TreeNodeTag`, which is quite unreliable.

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:555]: @beliefer can you fix it in a followup pr?

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:555]: @cloud-fan Thank you ping me.

**Pull request closed without merging**